File: Strings & 1D Lists/twoSum.py
# https://leetcode.com/problems/two-sum/

def twoSums(nums, target):
    numToIndex = {}
    for i, num in enumerate(nums):
        numToIndex[num] = i
    for i, num in enumerate(nums):
        if target-num in numToIndex and i != numToIndex[target-num]:
            return [i, numToIndex[target-num]]

File: Strings & Arrays/twoSum.go
func twoSum(nums []int, target int) []int {
    dict := map[int]int{};
    for i := 0; i < len(nums); i++ {
        dict[nums[i]] = i;
    }
    
    for i := 0; i < len(nums); i++ {
        index, ok := dict[target - nums[i]];
        if ok && i != index {
            return []int{i, index};
        }
    }
    return []int{};
}